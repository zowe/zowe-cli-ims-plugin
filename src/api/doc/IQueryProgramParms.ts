/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface representing parameters for the queryProgram API
 */
export interface IQueryProgramParms {
    /**
     * Specifies the 1-8 character name of the program.
     * Wildcards can be specified in the name.
     * The name is a repeatable parameter.
     * The default is NAME(*) which returns all program resources.
     */
    name?: string[];

    /**
     * Specifies the program output fields to be returned.
     * The program name is always returned, along with the name of the IMSâ„¢
     * that created the output, the region type, and the completion code.
     *
     * Available values : ALL, BMPTYPE, DEFN, DEFNTYPE, DOPT, FP, GLOBAL,
     * IMSID, GPSB, LANG, LOCAL, MODEL, RESIDENT, SCHDTYPE, STATUS,
     * TIMESTAMP, TRANSTAT, EXPORTNEEDED, DB, RTC, TRAN, WORK
     */
    attributes?: string[];

    /**
     * Selects programs for display that possess at least one of the
     * specified program status. This selection allows for additional
     * filtering by program status. The program status is returned
     * as output, even if the SHOW(STATUS) was not specified.
     *
     * Available values : DB-NOTAVL, IOPREV, LOCK, NOTINIT, STOSCHD, TRACE
     */
    status?: string[];

    /**
     *
     */
    route?: string[];

    /**
     * Index signature
     */
    [key: string]: any;
}
