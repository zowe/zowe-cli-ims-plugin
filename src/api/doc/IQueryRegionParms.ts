/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface representing parameters for the queryRegion API
 */
export interface IQueryRegionParms {
    /**
     * TODO The swagger doc is missing the description info
     * DC information
     * Available values : TRUE, FALSE (default TRUE)
     */
    dc?: boolean;

    /**
     * Include region information
     * Available values : TRUE, FALSE (default TRUE)
     */
    region?: boolean;

    /**
     *
     */
    route?: string[];

    /**
     * Index signature
     */
    [key: string]: any;
}
