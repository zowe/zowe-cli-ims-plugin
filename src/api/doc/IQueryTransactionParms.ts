/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface representing parameters for the queryTransaction API
 */
export interface IQueryTransactionParms {
    /**
     * TODO The swagger doc is missing the description info
     * Specifies the 1-8 character name of the transaction.
     * Wildcards can be specified in the name.
     * The name is a repeatable parameter.
     * The default is NAME(*) which returns all program resources.
     */
    name?: string[];

    /**
     * Specifies the transaction output fields to be returned.
     * The transaction name is always returned, along with the name of the IMSâ„¢
     * that created the output, the region type, and the completion code.
     *
     * AAvailable values : AFFIN, ALL, AOCMD, CLASS, CMTMODE, CONV, CPRI,
     * DCLWA, DEFN, DEFNTYPE, DIRROUTE, EDITRTN, EDITUC, EMHBSZ, EXPRTIME,
     * FP, GLOBAL, IMSID, INQ, LCT, LOCAL, LPRI, MAXRGN, MODEL, MSGTYPE,
     * MSNAME, NPRI, PARLIM, PGM, PLCT, PLCTTIME, PSB, QCNT, RECOVER,
     * REMOTE, RESP, RGC, SEGNO, SEGSZ, SERIAL, SPASZ, SPATRUNC, STATUS,
     * TIMESTAMP, TRANSTAT, WFI, WORK, EXPORTNEEDED
     */
    attributes?: string[];

    /**
     * Selects transactions for display that possess at least one of the
     * specified transaction status. This selection allows for additional
     * filtering by transaction status. The transaction status is returned
     * as output, even if the SHOW(STATUS) was not specified.
     *
     * Available values : AFFIN, BAL, CONV, CPIC, DYN, IOPREV, LCK,
     * NOTINIT, QERR, QSTP, SUSPEND, STOQ, STOSCHD, TRACE, USTO
     */
    status?: string[];

    /**
     *
     */
    route?: string[];

    /**
     *
     */
    class?: number[];

    /**
     *
     * Available values : LT, LE, GT, GE, EQ, NE
     */
    qcntcomp?: string[];

    /**
     *
     */
    qcntval?: number;

    /**
     *
     * Available values : N, Y
     */
    conv?: string;

    /**
     *
     * Available values : E, N, P
     */
    fp?: string;

    /**
     *
     * Available values : N, Y
     */
    remote?: string;

    /**
     *
     * Available values : N, Y
     */
    resp?: string;

    /**
     * Index signature
     */
    [key: string]: any;
}
