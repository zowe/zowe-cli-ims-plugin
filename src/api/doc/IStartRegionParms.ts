/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface representing parameters for the startRegion API
 */
export interface IStartRegionParms {

    /**
     * The name of the member containing JCL for the region to start.
     * The maximum length of the member name is eight characters.
     * If no member name is specified, the default
     * member name is used
     */
    memberName?: string;

    /**
     * If you specify the --local option, IMS overrides the symbolic IMSID parameter
     * in the JCL of the default or specified member. --local is the default if you specify
     * the --job-name option.
     */
    local?: boolean;

    /**
     * Use this option to override the job name on the JOB statement of the
     * default or specified JCL member for a dependent region
     */
    job_name?: string;

    /**
     * Route the stop region command to specific IMS regions
     */
    route?: string[];

    /**
     * Index signature
     */
    [key: string]: any;
}
