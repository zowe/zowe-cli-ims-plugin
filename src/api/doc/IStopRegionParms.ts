/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface representing parameters for the stopRegion API
 */
export interface IStopRegionParms {
    /**
     * Region Number Identifier. Can specify a Region Number or a Job Name, but NOT BOTH.
     */
    reg_num?: number[];

    /**
     * Can specify a Region Number or a Job Name, but NOT BOTH.
     */
    job_name?: string;

    /**
     * If the transaction indicated by this parameter is running in the specified region to stop,
     * an error message is received at the master terminal, indicating an application
     * program abend. The region will remain active, but the transaction will be
     * stopped.
     * The stop region command will be ignored if the transaction is not currently scheduled
     * in the specified region to stop
     * Specify a transaction to abnormally terminate
     */
    abdump?: string;

    /**
     * Specify a transaction in wait-for-input (WFI) mode to stop its message processing within the specified region
     */
    transaction?: string;

    /**
     * Is used if the region cannot be stopped with a stop region --abdump
     * command and must be preceded by a stop region --abdump command.
     */
    cancel?: boolean;

    /**
     * Route the stop region command to specific IMS regions
     */
    route?: string[];

    /**
     * Index signature
     */
    [key: string]: any;
}
