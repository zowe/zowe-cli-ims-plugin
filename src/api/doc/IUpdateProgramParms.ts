/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface representing parameters for the update program API
 */
export interface IUpdateProgramParms {
    /**
     * Specifies the 1-8 character name of the program.
     * Wildcards can be specified in the name.
     * name is a repeatable parameter.
     */
    name: string[];

    /**
     *
     */
    route?: string[];

    /**
     * Use this option to start the program(s)
     * Available values are SCHD, TRACE and REFRESH
     */
    start?: string[];

    /**
     * Use this option to stop the program(s)
     * Available values are SCHD and TRACE
     */
    stop?: string[];

    /**
     *
     */
    bmptype?: string;

    /**
     *
     */
    dopt?: string;

    /**
     *
     */
    fp?: string;

    /**
     *
     */
    gpsb?: string;

    /**
     *
     */
    lang?: string;
    /**
     *
     */
    lock?: string;

    /**
     *
     */
    resident?: string;

    /**
     *
     */
    schdtype?: string;

    /**
     *
     */
    transtat?: string;

    /**
     *
     */
    option?: string;

    /**
     * Index signature
     */
    [key: string]: any;
}
