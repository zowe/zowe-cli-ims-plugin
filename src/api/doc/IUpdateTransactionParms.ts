/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface representing parameters for the update transaction API
 */
export interface IUpdateTransactionParms {
    /**
     * Specifies the 1-8 character name of the transaction.
     * Wildcards can be specified in the name.
     * name is a repeatable parameter.
     */
    name: string[];

    /**
     *
     */
    route?: string[];

    /**
     *
     */
    class?: number[];

    /**
     *
     */
    option?: string;

    /**
     *
     */
    scope?: string[];

    /**
     * Use this option to start the program(s)
     * Available values are Q, SCHD, SUSPEND, and TRACE
     */
    start?: string[];

    /**
     * Use this option to stop the program(s)
     * Available values are Q, SCHD and TRACE
     */
    stop?: string[];

    /**
     *
     */
    aocmd?: string;

    /**
     *
     */
    setClass?: number;

    /**
     *
     */
    cmtmode?: string;

    /**
     *
     */
    conv?: string;

    /**
     *
     */
    cpri?: number;

    /**
     *
     */
    dclwa?: string;

    /**
     *
     */
    dirroute?: string;

    /**
     *
     */
    editrtn?: string;

    /**
     *
     */
    edituc?: string;

    /**
     *
     */
    emhbsz?: number;

    /**
     *
     */
    exprtime?: number;

    /**
     *
     */
    fp?: string;

    /**
     *
     */
    inq?: string;

    /**
     *
     */
    lct?: number;

    /**
     *
     */
    lpri?: number;

    /**
     *
     */
    lock?: string;

    /**
     *
     */
    maxrgn?: number;

    /**
     *
     */
    msgtype?: string;

    /**
     *
     */
    msname?: string;

    /**
     *
     */
    npri?: number;

    /**
     *
     */
    parlim?: number;

    /**
     *
     */
    pgm?: string;

    /**
     *
     */
    plct?: number;

    /**
     *
     */
    plcttime?: number;

    /**
     *
     */
    recover?: string;

    /**
     *
     */
    remote?: string;

    /**
     *
     */
    resp?: string;

    /**
     *
     */
    segno?: number;

    /**
     *
     */
    segsz?: number;

    /**
     *
     */
    serial?: string;

    /**
     *
     */
    sidl?: number;

    /**
     *
     */
    sidr?: number;

    /**
     *
     */
    spasz?: number;

    /**
     *
     */
    spatrunc?: string;

    /**
     *
     */
    transtat?: string;

    /**
     *
     */
    wfi?: string;

    /**
     * Index signature
     */
    [key: string]: any;
}
