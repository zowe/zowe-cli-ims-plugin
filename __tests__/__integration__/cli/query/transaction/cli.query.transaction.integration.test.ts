/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

// Test environment will be populated in the "beforeAll"
import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../../__src__/doc/ITestPropertiesSchema";

let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;
describe("ims query transaction", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "query_transaction_command",
            installPlugin: true,
            skipProperties: true
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should display the help", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/query_transaction_help.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });

});
