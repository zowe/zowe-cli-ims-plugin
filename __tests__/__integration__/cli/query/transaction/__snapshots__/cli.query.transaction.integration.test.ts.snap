// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ims query transaction should display the help 1`] = `
"===============IMS QUERY TRANSACTION HELP===============

 COMMAND NAME
 ------------

   transaction | tran

 DESCRIPTION
 -----------

   Query an IMS transaction.

 USAGE
 -----

   zowe ims query transaction [name...] [options]

 POSITIONAL ARGUMENTS
 --------------------

   name...		 (string)

      Specifies the name of transaction(s) to query. You can use an * character as a
      wildcard to select multiple transactions.

 OPTIONS
 -------

   --attributes  | --att (array)

      Specifies the transaction output fields to return.

      Allowed values: ALL, BMPTYPE, DEFN, DEFNTYPE, DOPT, FP, GLOBAL, IMSID, GPSB,
      LANG, LOCAL, MODEL, RESIDENT, SCHDTYPE, STATUS, TIMESTAMP, TRANSTAT,
      EXPORTNEEDED, DB, RTC, TRAN, WORK

   --status  | --st (array)

      Selects transactions that possess at least one of the specified transaction
      statuses.

      Allowed values: AFFIN, BAL, CONV, CPIC, DYN, IOPREV, LCK, NOTINIT, QERR, QSTP,
      SUSPEND, STOQ, STOSCHD, TRACE, USTO

   --route  | --rt (array)

      Specifies the routes to return.

   --class  | --cl (array)

      Selects transactions by the classes you specify.

   --queue-count-operator  | --qco (array)

      The compare operator used to select transactions based on queue count. Valid
      values: LT, LE, GT, GE, EQ or NE.

   --queue-count-value  | --qcv (number)

      The numeric value used with 'queue_count_operator' to select transactions based
      on queue count.

   --conversation-attributes  | --ca (string)

      Selects transactions by the conversational attributes you specify.

   --fast-path-options  | --fpo (string)

      Selects transactions by the Fast Path options you specify.

   --remote-option-specified  | --ros (string)

      Selects transactions by the remote option you specify.

   --response-mode-option-specified  | --rmos (string)

      Selects transactions by the response mode option you specify.

 IMS CONNECTION OPTIONS
 ----------------------

   --host  | -H (string)

      The IMS Operations API server host name.

   --port  | -P (number)

      The IMS Operations API server port.

   --ims-connect-host  | --ich (string)

      The hostname of your instance of IMS Connect. This is typically the hostname of
      the mainframe LPAR where IMS Connect is running.

   --ims-connect-port  | --icp (number)

      The port of your instance of IMS Connect. This port can be found in your IMS
      Connect configuration file on the mainframe.

   --plex  | -x (string)

      The name of the IMS plex.

   --user  | -u (string)

      The web server user name where the IMS Operations API resides.

   --password  | --pass (string)

      The web server user password where the IMS Operations API resides.

   --base-path  | --bp (string)

      The base path for your API mediation layer instance. Specify this option to
      prepend the base path to all z/OSMF resources when making REST requests. Do not
      specify this option if you are not using an API mediation layer.

 PROFILE OPTIONS
 ---------------

   --ims-profile  | --ims-p (string)

      The name of a (ims) profile to load for this command execution.

   --base-profile  | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --token-type  | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value  | --tv (string)

      The value of the token to pass to the API.

 RESPONSE FORMAT OPTIONS
 -----------------------

   --response-format-filter  | --rff (array)

      Filter (include) fields in the response. Accepts an array of field/property
      names to include in the output response. You can filter JSON objects properties
      OR table columns/fields. In addition, you can use this option in conjunction
      with '--response-format-type' to reduce the output of a command to a single
      field/property or a list of a single field/property.

   --response-format-type  | --rft (string)

      The command response output format type. Must be one of the following:

      table: Formats output data as a table. Use this option when the output data is
      an array of homogeneous JSON objects. Each property of the object will become a
      column in the table.

      list: Formats output data as a list of strings. Can be used on any data type
      (JSON objects/arrays) are stringified and a new line is added after each entry
      in an array.

      object: Formats output data as a list of prettified objects (or single object).
      Can be used in place of \\"table\\" to change from tabular output to a list of
      prettified objects.

      string: Formats output data as a string. JSON objects/arrays are stringified.

      Allowed values: table, list, object, string

   --response-format-header  | --rfh (boolean)

      If \\"--response-format-type table\\" is specified, include the column headers in
      the output.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a the group

   --help-web  | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Query transaction information for transaction named TRN12:

      $ zowe ims query transaction \\"TRN12\\"

   - Query transaction information for transactions named TRAN1
   and TRAN2:

      $ zowe ims query transaction \\"TRAN1 TRAN2\\"

   - Query transaction information for transactions starting
   with TRAN using the wild card character '*':

      $ zowe ims query transaction \\"TRAN*\\"

   - Query transaction information for all transactions (default
   is all):

      $ zowe ims query transaction

   - Query transaction information for all transactions
   specifying optional parameters:

      $ zowe ims query transaction --attributes \\"AFFIN TIMESTAMP\\" --status \\"NOTINIT\\" --route \\"IMS1 IMS2\\"

   - Query transaction information for all transactions
   specifying optional connection parameters:

      $ zowe ims query transaction --user \\"username\\" --pass \\"pass1234\\" --host \\"localhost\\" --port 8080 --ich \\"zos1\\" --icp 9999 --plex \\"PLEX1\\"

"
`;
